pkg/dogstatsd/enrich.go

```diff
@@ -48,7 +54,21 @@ func extractTagsMetadata(tags []string, defaultHostname string, originTags strin
 		k8sOrigin = kubelet.KubePodTaggerEntityPrefix + entityIDValue
 	}
 
-	return tags, host, origin, k8sOrigin
+	return tags, host, origin, k8sOrigin, taggerCardinality(cardinality)
+}
+
+func taggerCardinality(cardinality string) collectors.TagCardinality {
+	if cardinality == "" {
+		return tagger.DogstatsdCardinality
+	}
+
+	taggerCardinality, err := collectors.StringToTagCardinality(cardinality)
+	if err != nil {
+		log.Tracef("Couldn't convert cardinality tag: %w", err)
+		return tagger.DogstatsdCardinality
+	}
+
+	return taggerCardinality
```

Comment:
Performances are really tight in dogstatsd. Could we return a string from `extractTagsMetadata` and do this either in the tagger directly ?

it would also avoid the dependency to tagger/collectors everywhere.

Reply:
Actually `collectors.TagCardinality` is an `int`, sorry if I'm missing something obvious but could you elaborate why this can raise performance concerns?

In addition, regardless of the type, whether we run this in dsd or in the tagger it would have the same impact on dsd right? because dsd calls `tagger.TagBuilder` which is a blocking function. Not sure how moving this logic to the tagger could help.

In any case, the main reason why I've put this logic here is to avoid changing the tagger API which expects a `collectors.TagCardinality`

`TagBuilder(entity string, cardinality collectors.TagCardinality, tb *util.TagsBuilder) error`

Reply:
This is the loop that process dogstatsd packet. The slower it is the bigger are the packets drop. That's why we try to push as much as the post-processing later in the pipeline as we can.

Dogstatsd actually don't call the tagger anymore, the aggregator does. We move the hashing later to reduce allocation and GC run time.

We could do this in `enrichTag` to not change the tagger API then.

WDYT ?

Reply:
Thanks for the explanation, I have a clearer picture now! Moving it to `enrichTags` sounds good 👍

Reply:
PR updated